homeassistant/components/denonavr/__init__.py

```diff
@@ -33,3 +44,53 @@ def service_handler(service):
         hass.services.register(DOMAIN, service, service_handler, schema=schema)
 
     return True
+
+
+async def async_setup_entry(
+    hass: core.HomeAssistant, entry: config_entries.ConfigEntry
+):
+    """Set up the denonavr components from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+
+    # Connect to receiver
+    connect_denonavr = ConnectDenonAVR(
+        hass,
+        entry.data[CONF_HOST],
+        entry.data[CONF_TIMEOUT],
+        entry.data[CONF_SHOW_ALL_SOURCES],
+        entry.data[CONF_ZONE2],
+        entry.data[CONF_ZONE3],
+    )
+    if not await connect_denonavr.async_connect_receiver():
+        raise ConfigEntryNotReady
+    receiver = connect_denonavr.receiver
+
+    hass.data[DOMAIN][entry.entry_id] = receiver
+
+    device_registry = await dr.async_get_registry(hass)
+    device_registry.async_get_or_create(
+        config_entry_id=entry.entry_id,
+        connections={(dr.CONNECTION_NETWORK_MAC, entry.data[CONF_MAC])},
+        manufacturer=entry.data[CONF_MANUFACTURER],
+        name=entry.title,
+        model=f"{entry.data[CONF_MODEL]}-{entry.data[CONF_TYPE]}",
```

Comment:
Why do we use the receiver type as part of the model string? For my AVR-2100W I get `AVR-X2100W-avr-x`.

Reply:
Because that makes my job way easier ;)
There are 3 protocols used: avr, avr-x and avr-x-2016.
So if someone reports an issue, my first question will be to look in HomeAssistant and report the device model so that I now in which protocol the issue arises.

Unfortunately there is no separate property to specify protocol version in HomeAssistant

Reply:
I'm not sure if we should change the way the device data is displayed to the user. Information about the type of device will be in the log.